phantazm-mob/src/main/java/com/github/phantazmnetwork/mob/goal/FollowEntityGoal.java

```diff
@@ -0,0 +1,73 @@
+package com.github.phantazmnetwork.mob.goal;
+
+import com.github.phantazmnetwork.api.VecUtils;
+import com.github.phantazmnetwork.mob.PhantazmMob;
+import com.github.phantazmnetwork.mob.target.TargetSelectorInstance;
+import com.github.phantazmnetwork.mob.target.TargetSelector;
+import com.github.phantazmnetwork.neuron.bindings.minestom.entity.goal.NeuralGoal;
+import net.minestom.server.entity.Entity;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.Objects;
+
+/**
+ * A {@link Goal} that makes a {@link PhantazmMob} follow {@link Entity}s
+ * @param <TEntity> The type of {@link Entity} to follow
+ */
+public abstract class FollowEntityGoal<TEntity extends Entity> implements Goal {
+
+    private final @NotNull TargetSelector<TEntity> selectorCreator;
+
+    /**
+     * Creates a {@link FollowEntityGoal}.
+     * @param selector The {@link TargetSelector} used to select {@link Entity}s
+     */
+    public FollowEntityGoal(@NotNull TargetSelector<TEntity> selector) {
+        this.selectorCreator = Objects.requireNonNull(selector, "selector");
+    }
+
+    @Override
+    public @NotNull NeuralGoal createGoal(@NotNull PhantazmMob mob) {
+        TargetSelectorInstance<TEntity> selector = selectorCreator.createSelector(mob);
+
+        return new NeuralGoal() {
+            @Override
+            public boolean shouldStart() {
+                return true;
+            }
+
+            @Override
+            public void start() {
+                mob.entity().getNavigator().setDestination(() -> {
+                    return selector.selectTarget()
+                            .map(player -> VecUtils.toBlockInt(player.getPosition()))
```

Comment:
this should call MinestomDescriptor#computeTargetPosition (see NeuralEntity#setTarget(Entity)) so players don't become unreachable as soon as they jump

Reply:
or you can just call mob.entity().setTarget(player)